op-batcher/batcher/driver.go

```diff
@@ -324,6 +329,71 @@ func (l *BatchSubmitter) loop() {
 	}
 }
 
+// checkRecentTxsOnStart Check to see if there was a batcher tx sent recently that
+// still needs more block confirmations before being considered finalized
+func (l *BatchSubmitter) checkRecentTxsOnStart() {
```

Comment:
This requires archival node support

Reply:
We need to be mindful about requiring archival node support, i don't believe it was required previously but we shouldn't add the requirement unless its communicated that its now being added and everybody knows and its no issue to not use an archive node

Reply:
Great point. With this function we are only attempting to look back in time as far as the `currentBlockNum - confirmationDepth` which _should_ only be ~5 blocks behind the head of the L1 chain. If `confirmationDepth` is only set to 5, would that still require archival node support? Do regular nodes have any guarantees about how far back in time they will service `NonceAt` requests?

Reply:
Normal nodes usually have ~128 blocks of full state available

Reply:
Would it be reasonable and simpler to just wait for the op-node instance sync to reach the L1 head the batcher observes at startup? Then any already included transactions would be reflected in its state.  Checking the nonce can shortcircuit this if there are no recent transactions but with a typical follow distance of 5 blocks it seems reasonable to effectively wait 1 minute after startup to ensure the node is in sync.

Also my intuition (maybe I misunderstood) is that it's the follow distance of op-node that matters here, not the confirmation depth of the batcher. The batcher starts from the op-node `safe` head and submits blocks up to its `unsafe` head, so as long as the op-node has processed the batcher's previous transaction, the batcher won't try to resubmit those blocks.